### PR TITLE
Implementation of FurnaceConsumFuelEvent

### PR DESCRIPTION
**Pull request by Piratdu52**
Actually i can't test this because furnace.getData throw an abstract method error,  so in the state is more like a WIP don't hesitate to provide advice and require modifications.

### CODE REVIEW

**Comment by Grinch** [comment]
You may want to format your code a bit. As it stands, it is extremely difficult to read.

**Comment by Piratdu52** [comment]
I have checked your guidelines here, https://docs.spongepowered.org/en/contributing/guidelines.html and all seem okay. Maybe you want more vertical space inside my method ? If not please hilight an issue.

**Comment by twizmwazin** [comment]
The spacing is inconsistent. You have an extra space at line 96, and are omitting spaces around operators and curly brackets everywhere. I'm not sure if I'm missing something here, but why are you casting to Object before casting to the correct class in lines 84 and 85?

**Comment by Piratdu52** [comment]
Ok, I see. Will edit that later.

The (Object) is here for compilation purpose because I cast a net.micraft ItemStack to Sponge ItemStack or TileEntityFurnace to Furnace. Without I can't compile.

**Comment by ST-DDT** [comment]
Try using the auto formatter from your Ide.
In the extra folder (from the API) there are formatter files you can import and use.

File: src/main/java/org/spongepowered/common/mixin/core/block/tiles/MixinTileEntityFurnace.java [hunk 1]
```diff
@@ -56,4 +71,51 @@ public DataContainer toContainer() {
...
+                    remainingFuel= event.getRemainingFuel().get();
+                    remainingFuel.setQuantity(remainingFuel.getQuantity()+1); //add one because The mojang class will decrement this
+                }
+                else {
+                    ItemStackBuilder itemStackBuilder= Sponge.getGame().getRegistry().getItemBuilder();
+                    remainingFuel= itemStackBuilder.itemType(ItemTypes.SPONGE).quantity(1).build(); //set  a arbitrary itemStack with quantity=1,
+                                                                                                    // the Mojang class will delete it
+                }
+                this.furnaceItemStacks[1] = (net.minecraft.item.ItemStack) remainingFuel;
+                return;
```
Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/mixin/core/block/tiles/MixinTileEntityFurnace.java]: Did you mean for this to return from update()? If so, set `@Inject(cancellable=true)` and replace this with `ci.cancel()`.

**Comment by Zidane** [comment]
This event isn't correct anymore due to new events system.

**Pull request closed without merging**